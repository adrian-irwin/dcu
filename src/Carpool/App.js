import React, { useState } from "react";
import { StatusBar } from "expo-status-bar";
import { IconButton, Provider as PaperProvider } from "react-native-paper";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import LoginScreen from "./components/LoginScreen";
import PassengerHomePage from "./components/Passenger/HomePage";
import TransitionScreen from "./components/TransitionScreen";
import PassengerList from "./components/Driver/PassengerList";
import { LoginContext, CurrentUserContext } from "./components/Context";
import DriverHomePage from "./components/Driver/HomePage";
import RegisterScreen from "./components/RegisterScreen";
import PassengerInfo from "./components/Driver/PassengerInfo";
import PassengerMap from "./components/Passenger/Passenger";
import PassengerAdvertise from "./components/Passenger/AdvertiseForm";
import PassengerHistory from "./components/Passenger/History";

const Stack = createNativeStackNavigator();

export default function App() {
    const [loggedIn, setLoggedIn] = useState(false);
    const [currentUser, setCurrentUser] = useState({loggedIn: false});

    return (
        <LoginContext.Provider value={[loggedIn, setLoggedIn]}>
            <NavigationContainer>
                <PaperProvider>
                    <StatusBar style="auto" />
                    <CurrentUserContext.Provider value={[currentUser, setCurrentUser]}>
                        <Stack.Navigator
                            initialRouteName={loggedIn ? "Home" : "Login"}
                            screenOptions={{
                                headerShown: true,
                                headerTitleAlign: "center",
                            }}
                        >
                            {loggedIn ?
                                <Stack.Group
                                    screenOptions={{
                                        headerRight: () => (
                                            <IconButton
                                                icon="exit-to-app"
                                                size={25}
                                                onPress={() => {setLoggedIn(false); setCurrentUser({loggedIn: false});}}
                                            />
                                        )
                                    }}
                                >
                                    <Stack.Screen name="Home" component={TransitionScreen} options={{ headerTitle: "" }} />

                                    {/* Driver Pages */}
                                    <Stack.Screen name="DriverHomePage" component={DriverHomePage} options={{ headerTitle: "Driver Home" }} />
                                    <Stack.Screen name="PassengerList" component={PassengerList} options={{ headerTitle: "Passenger List" }} />
                                    <Stack.Screen name="PassengerInfo" component={PassengerInfo} options={{ headerTitle: "Passenger Info" }} />

                                    {/* Passenger Pages */}
                                    <Stack.Screen name="PassengerHomePage" component={PassengerHomePage} options={{ headerTitle: "Passenger Home" }} />
                                    <Stack.Screen name="PassengerMap" component={PassengerMap} options={{ headerTitle: "Passenger Map" }} />
                                    <Stack.Screen name="PassengerAdvertise" component={PassengerAdvertise} options={{ headerTitle: "Advertise Ride" }} />
                                    <Stack.Screen name="PassengerHistory" component={PassengerHistory} options={{ headerTitle: "Previous Rides" }} />
                                </Stack.Group>
                                :
                                <Stack.Group
                                    screenOptions={{
                                        headerTransparent: true,
                                        headerTitle: "Carpool2DCU",
                                    }}
                                >
                                    <Stack.Screen name="Login" component={LoginScreen} />
                                    <Stack.Screen name="Register" component={RegisterScreen} />
                                </Stack.Group>
                            }
                        </Stack.Navigator>
                    </CurrentUserContext.Provider>
                </PaperProvider>
            </NavigationContainer>
        </LoginContext.Provider>
    );
}
