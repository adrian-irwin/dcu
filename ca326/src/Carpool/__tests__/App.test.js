import React from "react";
import { fireEvent, render } from "@testing-library/react-native";

import App from "../App";

describe("App", () => {

    it("renders correctly", () => {
        const { toJSON } = render(<App />);
        expect(toJSON()).toMatchSnapshot();
    });

    it("renders Login screen by default", () => {
        const { getByText } = render(<App />);
        expect(getByText("Login")).toBeTruthy();
    });

    it("renders Register screen when Register button is pressed", () => {
        const { getByText } = render(<App />);
        const registerButton = getByText("Register");
        fireEvent.press(registerButton);
        expect(getByText("Register")).toBeTruthy();
    });

    it("renders email text input", () => {
        const { findByTestId } = render(<App />);
        expect(findByTestId("Email")).toBeTruthy();
    });

    it("renders password text input", () => {
        const { findByTestId } = render(<App />);
        expect(findByTestId("Password")).toBeTruthy();
    });
});
