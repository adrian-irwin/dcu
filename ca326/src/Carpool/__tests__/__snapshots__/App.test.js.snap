// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenStack
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          nativeBackButtonDismissalEnabled={false}
          onAppear={[Function]}
          onDisappear={[Function]}
          onDismissed={[Function]}
          onHeaderBackButtonClicked={[Function]}
          onNativeDismissCancelled={[Function]}
          onTransitionProgress={[Function]}
          onWillDisappear={[Function]}
          replaceAnimation="push"
          stackPresentation="push"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          swipeDirection="horizontal"
        >
          <RNSScreenStackHeaderConfig
            backButtonInCustomView={false}
            backgroundColor="transparent"
            color="rgb(0, 122, 255)"
            direction="ltr"
            disableBackButtonMenu={false}
            hidden={false}
            hideBackButton={false}
            hideShadow={true}
            largeTitleHideShadow={false}
            title="Carpool2DCU"
            titleColor="rgb(28, 28, 30)"
            topInsetEnabled={false}
            translucent={true}
          />
          <View
            accessibilityElementsHidden={false}
            importantForAccessibility="auto"
            style={
              Object {
                "flex": 1,
                "flexDirection": "column-reverse",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                  },
                  undefined,
                ]
              }
            >
              <View
                onLayout={[Function]}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#E7DCEB",
                      "flex": 1,
                      "justifyContent": "center",
                    },
                    Object {
                      "paddingBottom": 0,
                    },
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "width": "80%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "marginBottom": 5,
                      }
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 6,
                          },
                          Object {
                            "backgroundColor": "rgba(255, 251, 254, 1)",
                            "borderColor": "rgba(121, 116, 126, 1)",
                            "borderRadius": 4,
                            "borderWidth": 1,
                          },
                          undefined,
                        ]
                      }
                      testID="text-input-outline"
                    />
                    <View>
                      <View
                        style={
                          Array [
                            Object {
                              "paddingBottom": 0,
                            },
                            Object {
                              "minHeight": 56,
                              "paddingTop": 8,
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="none"
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                              ],
                              "zIndex": 3,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            pointerEvents="none"
                            style={
                              Object {
                                "backgroundColor": "rgba(255, 251, 254, 1)",
                                "bottom": 4,
                                "left": 10,
                                "maxHeight": 2,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 6,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                ],
                                "width": 12,
                              }
                            }
                          />
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              Object {
                                "backgroundColor": "rgba(255, 251, 254, 1)",
                                "color": "transparent",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 8,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 0,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 33,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                  Object {
                                    "scaleY": 1,
                                  },
                                ],
                                "width": -16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            Email
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            onLayout={[Function]}
                            style={
                              Object {
                                "color": "rgba(103, 80, 164, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 16,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 32,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                            testID="Email-label-active"
                          >
                            Email
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              Object {
                                "color": "rgba(73, 69, 79, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 16,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 32,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                            testID="Email-label-inactive"
                          >
                            Email
                          </Text>
                        </View>
                        <TextInput
                          autoCapitalize="none"
                          editable={true}
                          keyboardType="email-address"
                          maxFontSizeMultiplier={1.5}
                          multiline={false}
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder=" "
                          placeholderTextColor="rgba(73, 69, 79, 1)"
                          selectionColor="rgba(103, 80, 164, 1)"
                          style={
                            Array [
                              Object {
                                "margin": 0,
                                "zIndex": 1,
                              },
                              Object {
                                "height": 48,
                              },
                              Object {
                                "paddingBottom": 0,
                                "paddingTop": 0,
                              },
                              Object {
                                "color": "rgba(28, 27, 31, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "paddingHorizontal": 16,
                                "textAlign": "left",
                                "textAlignVertical": "center",
                              },
                              false,
                              Array [
                                Object {},
                              ],
                              undefined,
                            ]
                          }
                          testID="Email"
                          underlineColorAndroid="transparent"
                        />
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "marginBottom": 5,
                      }
                    }
                  >
                    <View
                      pointerEvents="none"
                      style={
                        Array [
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 6,
                          },
                          Object {
                            "backgroundColor": "rgba(255, 251, 254, 1)",
                            "borderColor": "rgba(121, 116, 126, 1)",
                            "borderRadius": 4,
                            "borderWidth": 1,
                          },
                          undefined,
                        ]
                      }
                      testID="text-input-outline"
                    />
                    <View>
                      <View
                        style={
                          Array [
                            Object {
                              "paddingBottom": 0,
                            },
                            Object {
                              "minHeight": 56,
                              "paddingTop": 8,
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="none"
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "transform": Array [
                                Object {
                                  "translateX": 0,
                                },
                              ],
                              "zIndex": 3,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            pointerEvents="none"
                            style={
                              Object {
                                "backgroundColor": "rgba(255, 251, 254, 1)",
                                "bottom": 4,
                                "left": 10,
                                "maxHeight": 2,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 6,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                ],
                                "width": 12,
                              }
                            }
                          />
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              Object {
                                "backgroundColor": "rgba(255, 251, 254, 1)",
                                "color": "transparent",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 8,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 0,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 33,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                  Object {
                                    "scaleY": 1,
                                  },
                                ],
                                "width": -16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            Password
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            onLayout={[Function]}
                            style={
                              Object {
                                "color": "rgba(103, 80, 164, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 16,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 32,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                            testID="Password-label-active"
                          >
                            Password
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              Object {
                                "color": "rgba(73, 69, 79, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 16,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 32,
                                "transform": Array [
                                  Object {
                                    "translateX": 0,
                                  },
                                  Object {
                                    "translateY": 0,
                                  },
                                  Object {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                            testID="Password-label-inactive"
                          >
                            Password
                          </Text>
                        </View>
                        <TextInput
                          autoCapitalize="none"
                          editable={true}
                          keyboardType="default"
                          maxFontSizeMultiplier={1.5}
                          multiline={false}
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder=" "
                          placeholderTextColor="rgba(73, 69, 79, 1)"
                          secureTextEntry={true}
                          selectionColor="rgba(103, 80, 164, 1)"
                          style={
                            Array [
                              Object {
                                "margin": 0,
                                "zIndex": 1,
                              },
                              Object {
                                "height": 48,
                              },
                              Object {
                                "paddingBottom": 0,
                                "paddingTop": 0,
                              },
                              Object {
                                "color": "rgba(28, 27, 31, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "paddingHorizontal": 16,
                                "textAlign": "left",
                                "textAlignVertical": "center",
                              },
                              false,
                              Array [
                                Object {},
                              ],
                              undefined,
                            ]
                          }
                          testID="Password"
                          underlineColorAndroid="transparent"
                        />
                      </View>
                    </View>
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "alignSelf": undefined,
                          "bottom": undefined,
                          "left": undefined,
                          "position": undefined,
                          "right": undefined,
                          "shadowColor": "#000",
                          "shadowOffset": Object {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0,
                          "shadowRadius": 0,
                          "top": undefined,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 0,
                              "width": 0,
                            },
                            "shadowOpacity": 0,
                            "shadowRadius": 0,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "backgroundColor": "rgba(103, 80, 164, 1)",
                              "borderColor": "transparent",
                              "borderRadius": 20,
                              "borderStyle": "solid",
                              "borderWidth": 0,
                              "marginTop": 10,
                              "minWidth": 64,
                              "width": 200,
                            }
                          }
                        >
                          <View
                            accessibilityRole="button"
                            accessibilityState={
                              Object {
                                "disabled": false,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "overflow": "hidden",
                                },
                                false,
                                Object {
                                  "borderRadius": 20,
                                },
                              ]
                            }
                            testID="button"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                numberOfLines={1}
                                selectable={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "rgba(28, 27, 31, 1)",
                                      "fontFamily": "System",
                                      "fontSize": 14,
                                      "fontWeight": "500",
                                      "letterSpacing": 0.1,
                                      "lineHeight": 20,
                                    },
                                    Object {
                                      "textAlign": "left",
                                    },
                                    Object {
                                      "writingDirection": "ltr",
                                    },
                                    Array [
                                      Object {
                                        "marginHorizontal": 16,
                                        "marginVertical": 9,
                                        "textAlign": "center",
                                      },
                                      false,
                                      Object {
                                        "marginHorizontal": 24,
                                        "marginVertical": 10,
                                      },
                                      undefined,
                                      false,
                                      Object {
                                        "color": "rgba(255, 255, 255, 1)",
                                        "fontFamily": "System",
                                        "fontSize": 14,
                                        "fontWeight": "500",
                                        "letterSpacing": 0.1,
                                        "lineHeight": 20,
                                      },
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                Login
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "alignSelf": undefined,
                          "bottom": undefined,
                          "left": undefined,
                          "position": undefined,
                          "right": undefined,
                          "shadowColor": "#000",
                          "shadowOffset": Object {
                            "height": 0,
                            "width": 0,
                          },
                          "shadowOpacity": 0,
                          "shadowRadius": 0,
                          "top": undefined,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          Object {
                            "shadowColor": "#000",
                            "shadowOffset": Object {
                              "height": 0,
                              "width": 0,
                            },
                            "shadowOpacity": 0,
                            "shadowRadius": 0,
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "borderColor": "transparent",
                              "borderRadius": 20,
                              "borderStyle": "solid",
                              "borderWidth": 0,
                              "marginTop": 10,
                              "minWidth": 64,
                              "width": 200,
                            }
                          }
                        >
                          <View
                            accessibilityRole="button"
                            accessibilityState={
                              Object {
                                "disabled": false,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Array [
                                Object {
                                  "overflow": "hidden",
                                },
                                false,
                                Object {
                                  "borderRadius": 20,
                                },
                              ]
                            }
                            testID="button"
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "justifyContent": "center",
                                  },
                                  undefined,
                                ]
                              }
                            >
                              <Text
                                numberOfLines={1}
                                selectable={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "rgba(28, 27, 31, 1)",
                                      "fontFamily": "System",
                                      "fontSize": 14,
                                      "fontWeight": "500",
                                      "letterSpacing": 0.1,
                                      "lineHeight": 20,
                                    },
                                    Object {
                                      "textAlign": "left",
                                    },
                                    Object {
                                      "writingDirection": "ltr",
                                    },
                                    Array [
                                      Object {
                                        "marginHorizontal": 16,
                                        "marginVertical": 9,
                                        "textAlign": "center",
                                      },
                                      false,
                                      Object {
                                        "marginHorizontal": 12,
                                      },
                                      undefined,
                                      false,
                                      Object {
                                        "color": "#1d1a29",
                                        "fontFamily": "System",
                                        "fontSize": 14,
                                        "fontWeight": "500",
                                        "letterSpacing": 0.1,
                                        "lineHeight": 20,
                                      },
                                      undefined,
                                    ],
                                  ]
                                }
                              >
                                Register
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenStack>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
